app/containers/Products/ProductRowItemContainer.js

```diff
@@ -81,37 +37,26 @@ export const ProductRowItemContainer = props => {
   };
 
   // Toggle the 'archived' value of a Product
-  const toggleArchived = event => {
+  const toggleArchived = async event => {
     event.preventDefault();
     event.stopPropagation();
-    const toggleArchived = props.product.state !== 'archived';
-    const saveVariables = {
+    const newState = props.product.state === 'archived' ? 'active' : 'archived';
+    const currentBenchmark = getCurrentBenchmark();
+    const variables = {
       input: {
-        product: {
-          name: props.product.name,
-          description: props.product.description,
-          state: toggleArchived ? 'archived' : 'active',
-          parent: [props.product.rowId]
-        }
+        newName: props.product.name,
+        newDescription: props.product.description || '',
+        newState,
+        prevId: props.product.rowId,
+        newParent: [props.product.rowId],
+        benchmarkId: currentBenchmark ? currentBenchmark.rowId : null
       }
     };
-    const {environment} = props.relay;
-    const saveMutation = createProduct;
-    commitMutation(environment, {
-      mutation: saveMutation,
-      variables: saveVariables,
-      onCompleted: async response => {
-        const currentBenchmark = getCurrentBenchmark();
-        const benchmarkPatch = {
-          productId: response.createProduct.product.rowId
-        };
-        await editProduct();
-        if (currentBenchmark) {
-          await editBenchmark(currentBenchmark.rowId, benchmarkPatch);
-        }
-      },
-      onError: err => console.error(err)
-    });
+    const response = await saveProductMutation(
+      props.relay.environment,
+      variables
+    );
+    console.log(response);
```

Comment:
```suggestion
```

Reply:
placeholder for logging / displaying errors to the user